tests/local.rs

```diff
@@ -0,0 +1,183 @@
+#[macro_use]
+extern crate serial_test_derive;
+extern crate fantoccini;
+extern crate futures_util;
+
+use fantoccini::{error, Client};
+
+mod common;
+
+
+fn sample_page_url(port: u16) -> String {
+    format!("http://localhost:{}/sample_page.html", port)
+}
+
+async fn goto(mut c: Client, port: u16) -> Result<(), error::CmdError> {
+    let url = sample_page_url(port);
+    c.goto(&url).await?;
+    let current_url = c.current_url().await?;
+    assert_eq!(url.as_str(), current_url.as_str());
+    c.close().await
+}
+
+async fn new_window(mut c: Client, port: u16) -> Result<(), error::CmdError> {
+    let url = sample_page_url(port);
+    c.goto(&url).await?;
+    c.new_window(false).await?;
+    let handles = c.windows().await?;
+    assert_eq!(handles.len(), 2);
+    c.close().await
+}
+
+async fn new_window_switch(mut c: Client, _port: u16) -> Result<(), error::CmdError> {
+    let handle_1 = c.window().await?;
+    c.new_window(false).await?;
+    let handle_2 = c.window().await?;
+    assert_eq!(
+        handle_1, handle_2,
+        "After creating a new window, the session should not have switched to it"
+    );
+
+    let all_handles = c.windows().await?;
+    let new_window_handle = all_handles
+        .into_iter()
+        .find(|handle| handle != &handle_1)
+        .expect("Should find a differing handle");
+
+    c.switch_to_window(new_window_handle).await?;
+
+    let handle_3 = c.window().await?;
+    assert_ne!(
+        handle_3, handle_2,
+        "After switching to a new window, the handle should differ now."
+    );
+
+    c.close().await
+}
+
+async fn new_tab(mut c: Client, port: u16) -> Result<(), error::CmdError> {
```

Comment:
Should we make this `new_tab_switch`, and have it test all the same things that `new_window_switch` does?